scripts/UpdateRepos.ps1

```diff
@@ -56,11 +56,12 @@ Push-Location $ModuleDirectory
 try {
 
     $build_errors = @()
-    # Get-Submodules also update --init's them
     $submodules = Get-Submodules $RepoRoot
     foreach ($submodule in $submodules) {
         Push-Location $submodule.path
         try {
+            Invoke-Block { & git fetch }
+            Invoke-Block { & git checkout origin/$($submodule.branch) }
```

Comment:
This creates a branch named "origin/dev" locally, right? Might be better to do this:
```
git fetch origin dev:dev
git checkout dev
```

Reply:
It does not (as long as your remote is named origin). It puts you in a detached head state at HEAD of origin/dev. I prefer this method because it doesn't pollute the sub-module with a dev branch, so it's "cleaner" for when we have to run this locally. I've previously gotten into weird states with branches on sub-modules of Universe which have been annoying.

Reply:
ok, makes sense